### PR TITLE
question_text() uses textAreaInput() if rows or cols are provided

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes https://github.com/rstudio/learnr/issues/455

This PR adds `rows` and `cols` as parameters of `question_text()`. When provided, the text input UI will use `textAreaInput()` instead of `textInput()`.

I used `rows` and `cols` as the parameter names so we can inherit documentation from `textAreaInput()`, but I'd be happy to update the parameter names or documentation to specifically mention that they change the input UI.

### Reprex

Here's an example to try to create an "essay" input as described in #455. The question will always be graded as incorrect, but by setting `allow_retry = TRUE` and changing the `try_again_button` text, the input feels less binary correct/incorrect.

````
---
title: "Tutorial"
output: learnr::tutorial
runtime: shiny_prerendered
---

```{r setup, include=FALSE}
library(learnr)
```

## Text Questions

```{r question-text}
question_text("What's the best number of them all?", answer("42", TRUE))
```

```{r question-text-area}
question_text(
  "Why?", 
  answer("", correct = TRUE), # Question will always be "incorrect"
  rows = 5, 
  correct = NULL,
  incorrect = NULL,
  allow_retry = TRUE,
  try_again_button = "Edit Answer"
)
```
````

PR task list:
- [x] Update NEWS
- [x] Add tests (if possible)
- [x] Update documentation with `devtools::document()`

### CODE REVIEW

File: R/question_text.R [hunk 1]
```diff
@@ -58,7 +63,16 @@ question_text <- function(
...
 question_ui_initialize.learnr_text <- function(question, value, ...) {
-  textInput(
+  # Use textInput() unless one of rows or cols are provided
+  textInputFn <- if (is.null(question$options$rows) && is.null(question$options$cols)) {
+    textInput
+  } else {
+    function(...) {
+      textAreaInput(..., cols = question$options$cols, rows = question$options$rows)
+    }
+  }
```
Comment (schloerke) [inline R/question_text.R:73]:
```suggestion
  textInputFn <- 
    if (is.null(question$options$rows) && is.null(question$options$cols)) {
      textInput
    } else {
      function(...) {
        textAreaInput(..., cols = question$options$cols, rows = question$options$rows)
      }
    }
```

**Comment by schloerke** [comment]
cc @dtkaplan @davidkane9

Does this PR address https://github.com/rstudio/learnr/issues/455?

I think we'll hold off on a dedicated `learnr_text_area()` function for now.  If you find yourself using a consistent set of rows/cols... please open a new issue and we can open the discussion to add `learnr_text_area()` and default to `learnr_text_area(rows = m, cols = n)`

Thank you! and Thank you @gadenbuie!

File: man/question_text.Rd [hunk 2]
```diff
@@ -42,6 +44,16 @@ option.}
...
 
+\item{rows}{The value of the visible character rows of the input, e.g. \code{6}.
+If the \code{height} argument is specified, \code{height} will take precedence in the
+browser's rendering.}
+
+\item{cols}{Value of the visible character columns of the input, e.g. \code{80}.
+This argument will only take effect if there is not a CSS \code{width} rule
+defined for this element; such a rule could come from the \code{width} argument
+of this function or from a containing page layout such as
+\code{\link[shiny:fluidPage]{fluidPage()}}.}
```
Comment (schloerke) [inline man/question_text.Rd:55]: Let's go ahead and use custom parameter text.  Both of these talk about parts of a shiny UI that are hidden from tutorial authors.  Good catch!

**Comment by dtkaplan** [comment]
Yes, this does the job. Thanks!

File: man/question_text.Rd (hunk 2 above)
Comment (gadenbuie) [reply man/question_text.Rd]: Done!

**Pull request merged**